app/Http/Controllers/BuildController.php

```diff
@@ -858,24 +837,31 @@ public function viewFiles(): View|RedirectResponse
                     $ext = 'Tb';
                 }
 
-                $xml .= '<filesizedisplay>' . round($filesize) . ' ' . $ext . '</filesizedisplay>';
-                $xml .= '<isurl>' . $uploadFileOrURL->IsUrl . '</isurl>';
-                $xml .= '</uploadfile>';
-            } else {
-                $xml .= '<uploadurl>';
-                $xml .= '<id>' . $uploadFileOrURL->Id . '</id>';
-                $xml .= '<filename>' . htmlspecialchars($uploadFileOrURL->Filename) . '</filename>';
-                $xml .= '</uploadurl>';
+                $file['filesizedisplay'] = round($filesize) . ' ' . $ext;
+
+                $files[] = $file;
             }
         }
 
-        $xml .= '</cdash>';
+        return view('build.files')
+            ->with('build', $this->build)
+            ->with('files', $files)
+            ->with('urls', $urls);
+    }
 
-        return view('cdash', [
-            'xsl' => true,
-            'xsl_content' => generate_XSLT($xml, base_path() . '/app/cdash/public/viewFiles', true),
-            'title' => 'View Files'
-        ]);
+    public function build_file(int $build_id = null, int $file_id = null) : StreamedResponse
```

Comment:
Just like the other function, there is no way for this function to be called without parameters being specified.  You can remove the default values here.
```suggestion
    public function build_file(int $build_id, int $file_id) : StreamedResponse
```